Cargo.toml

```
@@ -7,8 +7,12 @@ license = "MIT/Apache-2.0"
 repository = "https://github.com/frewsxcv/rust-dark-light"
 description = "Detect if dark mode or light mode is enabled"
 readme = "README.md"
+build = "build.rs"
 
 [dependencies]
 
 [target.'cfg(windows)'.dependencies]
 winreg = "0.8.0"
+
+[target.'cfg(target_os = "macos")'.dependencies]
```

```suggestion
[target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies]
```

AFAIK this should work on iOS, right?

---

No, iOS APIs are completely different.

---

Are you sure? https://developer.apple.com/documentation/uikit/uitraitcollection/1651063-userinterfacestyle

---

Yeah, completely different API in a completely different framework.

That said, re-reading this thread - we might be taking "this should work on iOS" differently. You're talking about _this entire PR_, or just the `cfg` line? If the latter, then yeah, you're correct and I retract my comments there.

That said, the inner workings of the PR would need to change to be the right iOS calls, so I don't know if it makes sense to compile anything on iOS that doesn't have the extra changes necessary.

---

Would it be feasible and correct to use the `userInterfaceStyle` API like the old implementation? I'm not clear what the differences are.

---

The "best" thing here is probably something like...

- Keep the `objc` cfg macOS && iOS line
- Keep `macos.rs` file as-is, in this PR
- Link `UIKit` in `build.rs` if you're targeting iOS
- Add an `ios.rs` that calls `msg_send![class!(UIScreen) mainScreen] -> traitCollection -> userInterfaceStyle` and checks for the `UIUserInterfaceStyleDark ` value (whatever it is, I'd have to skim through the headers to look it up)

The thing to keep in mind here is that if something higher up in the chain forces `overrideUserInterfaceStyle`, I believe this can be out of sync with what the actual expected value is - on iOS, you would generally read this from your view-controller trait collection. This is why I'm saying it's fairly different from macOS's approach.

Ultimately, on iOS I imagine you'd (more often than not) generally be somewhere in a view chain already due to how software is written, and I would imagine this library wouldn't get much use there. As it is I marked this to myself awhile ago as something I'd need to shim differently in cacao so that lifecycle patterns fit better).

I also, off the top of my head, don't know how the `[UIScreen mainScreen]` light/dark mode return value would work with multiple screens (e.g iPadOS). Hence why I'd sooner just punt iOS altogether, but your call(s).

---

Okay, seems like it would be quite a bit of work to get this working on iOS, so yeah, no need to bother with that in this PR.